activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb

```diff
@@ -614,6 +603,14 @@ def connect
           )
         end
 
+        def reconnect
+          if active?
+            @raw_connection.rollback rescue nil
```

Comment:
There's no base class we could rescue here? Inline `rescue` like this always make me shiver.

Reply:
The one is also just a relocation (https://github.com/rails/rails/pull/44576/files#diff-13f18e985e903996e81f573b1fd364c19bddde279bc2797181491e6c146bcdc7L169), but yeah, probably worth reconsideration while we're passing by.

Reply:
I decided to leave this for now... as noted, it's just relocating existing code, and I'd rather not introduce an accidental behaviour change by omitting an exception it should rescue.

We should still revisit, it just exceeded my added-risk appetite for this PR when I sat down to enumerate reasonably-rescuable exceptions.